### PR TITLE
feat: add manually install harvester cloud provider in rke2

### PR DESCRIPTION
**Pull request by FrankYang0529**

### CODE REVIEW

File: docs/rancher/cloud-provider.md [hunk 1]
```diff
@@ -63,6 +63,101 @@ Starting with Rancher v2.9.0, you can configure a specific folder for cloud conf
...
+      permissions: '0644'
+    ```
+
+1. Configure the **Cloud Provider** either to **External**.
+    ![](/img/v1.4/rancher/external-harvester-cloud-provider.png)
+
+1. Copy and paste the `cloud-init user data` content to **Machine Pools >Show Advanced > User Data**.
+    ![](/img/v1.2/rancher/cloud-config-userdata.png)
+
+1. Add the following `HelmChart` yaml of `harvester-cloud-provider` to **Cluster Configuration > Add-On Config > Additional Manifest**. Remember to change `<cluster-name>` to the name of your cluster.
```
Comment (jillian-maroket) [inline docs/rancher/cloud-provider.md:135]:
```suggestion
1. On the Harvester UI, go to the **Cluster Configuration** screen and set the value of **Cloud Provider** to **External**.

    ![](/img/v1.4/rancher/external-harvester-cloud-provider.png)

1. Copy and paste the `cloud-init user data` content to **Machine Pools** > **Show Advanced** > **User Data**.

    ![](/img/v1.2/rancher/cloud-config-userdata.png)

1. Add the `HelmChart` CRD for `harvester-cloud-provider` to **Cluster Configuration** > **Add-On Config** > **Additional Manifest**. 

    You must replace `<cluster-name>` with the name of your cluster.
```

File: docs/rancher/cloud-provider.md [hunk 2]
```diff
@@ -63,6 +63,101 @@ Starting with Rancher v2.9.0, you can configure a specific folder for cloud conf
...
+      valuesContent: |-
+        global:
+          cattle:
+            clusterName: <cluster-name>
+    ```
+
+    ![](/img/v1.2/rancher/external-cloud-provider-addon.png)
+
+1. To create load balancer, please add `cloudprovider.harvesterhci.io/ipam: <dhcp|pool>` annotation.
+    ![](/img/v1.4/rancher/harvester-cloud-provider-loadbalancer-annotation.png)
```
Comment (jillian-maroket) [inline docs/rancher/cloud-provider.md:159]:
```suggestion
1. To create the load balancer, add the annotation `cloudprovider.harvesterhci.io/ipam: <dhcp|pool>`.

    ![](/img/v1.4/rancher/harvester-cloud-provider-loadbalancer-annotation.png)
```

File: docs/rancher/cloud-provider.md [hunk 3]
```diff
@@ -63,6 +63,101 @@ Starting with Rancher v2.9.0, you can configure a specific folder for cloud conf
 
   ![](/img/v1.4/rancher/rke2-cloud-provider-custom-data-dir.png)
 
+### Manually install in the RKE2 Cluster
```
Comment (jillian-maroket) [inline docs/rancher/cloud-provider.md:66]:
```suggestion
### Manually Deploying to the RKE2 Cluster
```

**Review by jillian-maroket — COMMENTED** [review]
Let me know if you have concerns about the changes.

**Comment by FrankYang0529** [comment]
Hi @jillian-maroket, thanks for review. I addressed all comments. 👍

**Review by jillian-maroket — APPROVED** [review]
LGTM. Thanks!

File: docs/rancher/cloud-provider.md [hunk 4]
```diff
@@ -63,6 +63,106 @@ Starting with Rancher v2.9.0, you can configure a specific folder for cloud conf
...
+    ########## cloud-init user data ############
+    write_files:
+    - encoding: b64
+      content: <CONTENT>
+      owner: root:root
+      path: /etc/kubernetes/cloud-config
+      permissions: '0644'
+    ```
+
+1. On the Harvester UI, go to the **Cluster Configuration** screen and set the value of **Cloud Provider** to **External**.
```
Comment (TachunLin) [inline docs/rancher/cloud-provider.md:129]: LGTM. 
Just a small suggestion whether we can use a more specific term for the `Harvester UI`, for example `Rancher UI` or `RKE2 cluster creation page`.

**Review by TachunLin — APPROVED** [review]
LGTM. 
Thank you for bringing up the manually deploy cloud provider guide for RKE2 cluster

File: docs/rancher/cloud-provider.md (hunk 4 above)
Comment (FrankYang0529) [reply docs/rancher/cloud-provider.md]: Hi @TachunLin, yeah, it's clearer for users. Thanks.

Comment (TachunLin) [reply docs/rancher/cloud-provider.md]: Thanks @FrankYang0529 for the update :+1:

**Pull request merged**